apps/web/app/(app)/environments/[environmentId]/surveys/PreviewSurvey.tsx

```diff
@@ -19,13 +20,16 @@ interface PreviewSurveyProps {
   activeQuestionId?: string | null;
   questions: Question[];
   brandColor: string;
+  highlightBorderColor?: string | null;
+  overwriteBorderHighlight?: boolean;
   environmentId: string;
   surveyType: Survey["type"];
   thankYouCard: Survey["thankYouCard"];
   autoClose: Survey["autoClose"];
   previewType?: "modal" | "fullwidth" | "email";
   product: TProduct;
   environment: TEnvironment;
+  overwritePosition?: PlacementType | null;
```

Comment:
I think it's easier to pass the whole `survey` to the `PreviewSurvey` component instead of passing every single key needed. Please rewrite to use `survey` as a prop 😊

Reply:
Need some help with this

if I replace all these with survey
![image](https://github.com/formbricks/formbricks/assets/69904519/1dab1071-186e-4fd2-b98f-4b076e4b46fe)

I had to make some of this 👇 props optional is that a right way to do it. 
![image](https://github.com/formbricks/formbricks/assets/69904519/2e5fcfde-88e4-4855-b8b2-28dc3766ea0c)

since this component is used in `TemplateContainer` component
![image](https://github.com/formbricks/formbricks/assets/69904519/99e3e7d8-d4e9-4222-96c3-889fdd58bbb5)